src/main/java/com/streamr/client/StreamrClient.java

```diff
@@ -229,7 +229,7 @@ private void connect(boolean firstTrial) throws ConnectionTimeoutException {
             } catch (InterruptedException e) {
                 e.printStackTrace();
             }
-            connect(false);
+            connect();
```

Comment:
If I understand correctly, because of this recursion, this method will actually block the caller until a connection is established. Could be a problem down the line?